lib/helpers/hearing_renderer.rb

```diff
@@ -0,0 +1,349 @@
+    details = []
+    details += [vet.name, "FN: #{vet.file_number}"] if show_pii
+    details << "PID: #{vet.participant_id}"
+    details
+  end
+
+  def readable_date(date, include_time = true)
+    if include_time
+      date.strftime("%m-%d-%Y %I:%M%p %Z")
+    else
+      date.strftime("%m-%d-%Y")
+    end
+  end
+
+  def veteran_children(vet)
+    appeals = Appeal.where(veteran_file_number: vet.file_number)
+    appeals.sort_by { |appeal| appeal.receipt_date || Time.zone.today }
+    legacy_appeals = LegacyAppeal.fetch_appeals_by_file_number(vet.file_number)
+    legacy_appeals.sort_by { |la| la.created_at || Time.zone.today }
+
+    children = []
+    children += legacy_appeals.map { |legacy_appeal| structure(legacy_appeal) }
+    children += appeals.map { |appeal| structure(appeal) }
+    children
+  end
+
+  def get_appeal_type_conversions(appeal)
+    unformatted_versions = appeal.versions
+    versions = unformatted_versions.map do |v|
+      change = v.changeset["changed_hearing_request_type"] ||
+               v.changeset["changed_request_type"]
+      {
+        "from_type" => print_nil(Hearing::HEARING_TYPES[change[0]&.to_sym]),
+        "to_type" => Hearing::HEARING_TYPES[change[1]&.to_sym],
+        "converted_by" => User.find(v.whodunnit).css_id,
+        "converted_at" => readable_date(v.created_at)
+      }
+    end
+
+    versions
+  end
+
+  def format_original_and_current_type(appeal)
+    original = print_nil(appeal.readable_original_hearing_request_type)
+    current = print_nil(appeal.readable_current_hearing_request_type)
+    if original == current && appeal.versions.count == 0
+      ["Current type: #{current}"]
+    else
+      ["Original Type: #{original}, current type: #{current}"]
+    end
+  end
+
+  def format_conversions(type_conversions)
+    type_conversions.map do |tc|
+      "Converted to #{tc['to_type']} from #{tc['from_type']} by #{tc['converted_by']} at #{tc['converted_at']}"
+    end
+  end
+
+  def add_original_type(original_type, type_conversions)
+    first_conversion = type_conversions[0]
+    first_conversion["from_type"] = original_type
+    type_conversions[0] = first_conversion
+
+    type_conversions
+  end
+
+  def get_appeal_history(appeal)
+    if appeal.versions.empty?
+      format_original_and_current_type(appeal)
+    else
+      type_conversions = get_appeal_type_conversions(appeal)
+      type_conversions_with_original_type = add_original_type(
+        appeal.readable_original_hearing_request_type,
+        type_conversions
+      )
+      text = format_original_and_current_type(appeal)
+      text += format_conversions(type_conversions_with_original_type)
+      text
+    end
+  end
+
+  def get_unscheduled_hearings(appeal)
+    unscheduled = appeal.tasks.open.of_type(:ScheduleHearingTask)&.pluck(:id)&.map { |i| "id: #{i}" }
+    # TODO, does this get the regional_office correctly?
+    regional_office = appeal.closest_regional_office || appeal.regional_office&.key
```

Comment:
let's replicate this [logic](https://github.com/department-of-veterans-affairs/caseflow/blob/470e570b62ec02bf0e39884a0a0e963f4ba1a338/app/services/cached_appeal_service.rb#L150) because that's what we used to populate the queue
```suggestion
    regional_office = appeal.closest_regional_office || COPY::UNKNOWN_REGIONAL_OFFICE
```